overlord/devicestate/devicestate.go

```diff
@@ -433,7 +435,47 @@ func (ms *modelSnapsForRemodel) canHaveUC18PinnedTrack() bool {
 		(ms.newModelSnap.SnapType == "kernel" || ms.newModelSnap.SnapType == "gadget")
 }
 
-func remodelEssentialSnapTasks(ctx context.Context, st *state.State, ms modelSnapsForRemodel, deviceCtx snapstate.DeviceContext, fromChange string) (*state.TaskSet, error) {
+type remodelOptions struct {
+	localSnapsRequired bool
+}
+
+func (ro *remodelOptions) UpdateWithDeviceContext(st *state.State,
```

Comment:
I may misunderstand the use-case/thoughts behind this, but I think this is a bit of a weird use-case of an Options structure. Why are these functions a part of this structure? They only use one member from the structure.

Reply:
I still agree with myself with that plan, but maybe in the context of this discussion remodelVariant could be a better name for the struct (I mused about it with myself before)

Reply:
Name changed now

Reply:
I think `remodelVariant` is nice!